src/task.c

```diff
@@ -558,6 +558,14 @@ JL_DLLEXPORT void jl_switchto(jl_task_t **pt)
 
 JL_DLLEXPORT JL_NORETURN void jl_no_exc_handler(jl_value_t *e)
 {
+    // NULL exception objects are used to denote the top of the exception stack.
```

Comment:
`excstack->top` is one off the end so that an empty stack has `excstack->top == 0`.  But the top might not be NULL; it's just junk data, IIRC.